src/search.c

```diff
@@ -5139,6 +5140,167 @@ fuzzy_match_str_with_pos(char_u *str UNUSED, char_u *pat UNUSED)
 #endif
 }
 
+/*
+ * This function searches for a fuzzy match of the pattern `pat` within the
+ * line pointed to by `*ptr`. It splits the line into words, performs fuzzy
+ * matching on each word, and returns the length and position of the first
+ * matched word.
+ */
+    static int
+fuzzy_match_str_in_line(char_u **ptr, char_u *pat, int *len, pos_T *current_pos)
+{
+    char_u	*str = *ptr;
+    char_u	*strBegin = str;
+    char_u	*end = NULL;
+    char_u	*start = NULL;
+    int		found = FALSE;
+    char_u	*word = NULL;
+    int		result;
+
+    if (str == NULL || pat == NULL)
+        return found;
+
+    while (*str != NUL)
+    {
+	// Skip non-word characters
+	start = find_word_start(str);
+	if (*start == NUL)
+	    break;
+	end = find_word_end(start);
+
+	// Extract the word from start to end
+	word = vim_strnsave(start, (int)(end - start));
```

Comment:
I don't think it's needed to allocate a string here, this should work just the same (pls somebody correct me if I'm wrong)

```c
	// Store the character after word, then make it NUL
	// so that fuzzy_match_str stops right there
	char oldchar = *end;
	*end = NUL;

	// Perform fuzzy match
	result = fuzzy_match_str(start, pat);

	// Restore the character we made NUL before
	*end = oldchar;
```

At least it seems to work just fine for me. There's a hell lot of allocations being done there, otherwise.

Reply:
yup this is effective.